src/graph/UserExecutor.cpp

```diff
@@ -0,0 +1,185 @@
+/* Copyright (c) 2020 vesoft inc. All rights reserved.
+ *
+ * This source code is licensed under Apache 2.0 License,
+ * attached with Common Clause Condition 1.0, found in the LICENSES directory.
+ */
+
+#include "common/encryption/MD5Utils.h"
+#include "graph/UserExecutor.h"
+
+namespace nebula {
+namespace graph {
+
+CreateUserExecutor::CreateUserExecutor(Sentence *sentence,
+                                       ExecutionContext *ectx)
+    : Executor(ectx, "create_user") {
+    sentence_ = static_cast<CreateUserSentence*>(sentence);
+}
+
+Status CreateUserExecutor::prepare() {
+    DCHECK(sentence_ != nullptr);
+    return Status::OK();
+}
+
+void CreateUserExecutor::execute() {
+    auto *mc = ectx()->getMetaClient();
+    auto* account = sentence_->getAccount();
+    auto opts = sentence_->getOpts();
+    auto* loginType = sentence_->getLoginType();
+    nebula::cpp2::UserItem userItem;
+    userItem.set_account(*account);
+    auto type = loginType->getLoginType() == UserLoginType::PASSWORD
+                ? nebula::cpp2::UserLoginType::PASSWORD
+                : nebula::cpp2::UserLoginType::LDAP;
+    userItem.set_login_type(type);
+    if (type == nebula::cpp2::UserLoginType::PASSWORD) {
+        userItem.set_encoded_pwd(encryption::MD5Utils::md5Encode(*loginType->getPassword()));
+    }
+    UserUtils::resetUserItem(opts, userItem);
+    auto future = mc->createUser(userItem, sentence_->ifNotExists());
+    auto *runner = ectx()->rctx()->runner();
+    auto cb = [this] (auto &&resp) {
+        if (!resp.ok()) {
+            DCHECK(onError_);
+            onError_(resp.status());
+            return;
+        }
+
+        DCHECK(onFinish_);
+        onFinish_(Executor::ProcessControl::kNext);
+    };
+
+    auto error = [this] (auto &&e) {
+        LOG(ERROR) << "Exception caught: " << e.what();
+        onError_(Status::Error("Internal error"));
+    };
+
+    std::move(future).via(runner).thenValue(cb).thenError(error);
+}
+
+DropUserExecutor::DropUserExecutor(Sentence *sentence,
+                                   ExecutionContext *ectx)
+    : Executor(ectx, "drop_user") {
+    sentence_ = static_cast<DropUserSentence*>(sentence);
+}
+
+Status DropUserExecutor::prepare() {
+    DCHECK(sentence_ != nullptr);
+    return Status::OK();
+}
+
+void DropUserExecutor::execute() {
+    auto *mc = ectx()->getMetaClient();
+    auto *account = sentence_->getAccount();
+    auto future = mc->dropUser(*account, sentence_->ifExists());
```

Comment:
Do we need to check the user privilege? I guess only user like GOD can drop/alter user

Reply:
> Do we need to check the user privilege? I guess only user like GOD can drop/alter user

Aha, Good question. Only GOD and ADMIN are allowed to create, delete and drop users. and GOD allows authorize for any space. ADMIN only allows authorize space within own permission.

And I will coding the privilege check function in next PR. the next PR is "Permission Manager"

Reply:
👍